dlrover/python/master/node/dist_job_manager.py

```diff
@@ -602,6 +600,8 @@ def _process_list_nodes(self, nodes: List[Node]):
                     )
                     new_node = copy.deepcopy(node)
                     self._job_context.update_job_node(new_node)
+                    if new_node.has_group():
+                        self._job_context.update_job_node_by_group(new_node)
```

Comment:
the new_node will be deep copy again in method 'update_job_node_by_group'

suggestion:
```
if node.has_group():
    self._job_context.update_job_node_by_group(node)
else:
    new_node = copy.deepcopy(node)
    self._job_context.update_job_node(new_node)
```